app/routes/routes.py

```diff
@@ -20,23 +49,90 @@ def FAQ_site():
     return render_template("main/FAQ.html")
 
 
-"""
-@app.route("/user_homepage", methods=["GET", "POST"])
+@app.route("/user_homepage", methods=["GET"])
+@login_required
 def user_homepage():
-    logout_user()
-    return redirect(url_for("home"))
-"""
+    robots_list = get_robots_for_user(current_user)
+    if not robots_list:
+        generate_random_robots(current_user, number_of_robots=5000)

```

Comment:
The function could just return the list 🤔 
A comment here might be helpful since if it would be a real application you wound't generate those robots